### PR TITLE
refactor: Qdrant vector database implementation

### PR DESCRIPTION
**Pull request by Anush008**
This PR
- Refactors the Qdrant client implementation to enable the use of an in-memory Qdrant instance, provide API keys and other params when instantiating Qdrant.
- Adds optional unit tests for the Qdrant implementation.

### CODE REVIEW

**Comment by doberst** [comment]
@Anush008 - really good stuff here - **thank you** - please give us a little time to review -  will also think about whether we should fully conform these changes across other vector db configs, etc.

**Comment by doberst** [comment]
@Anush008 - we have reviewed the change and are good with it - it is a nice improvement - we will look to do similar enhancements (with more config options) for the other vector db too.   The only issue is that we would like to keep the imports for pgvector, redis and qdrant 'optional' for now - and keep try/except "pass" rather than throwing an import error.    We are still debating how many of these to include in our pip install package - and want to keep any dependencies outside of the pip package as optional.   Welcome your thoughts on this, e.g., include more or less in pip.   If you could resubmit the pull request keeping the 'pass' if the import fails, then we will go ahead and merge.

File: llmware/embeddings.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 try:
     from pgvector.psycopg import register_vector
     import psycopg
-except:
+except ImportError:
     pass
```
Comment (Anush008) [inline llmware/embeddings.py:33]: @doberst, the pass is still in place. I've only replaced the bare except with `except ImportError`.

Comment (doberst) [reply llmware/embeddings.py:33]: Got it.   :).  My bad - should have had my second cup of coffee first ... 😄

**Comment by doberst** [comment]
Nice work - well done!

**Pull request merged**